internal/handlers/create_catalog.go

```diff
@@ -75,7 +76,7 @@ func (h *CreateCatalogHandler) Handle(message messaging.Message) error {
 		return fmt.Errorf("cannot discover repositories: %w", err)
 	}
 
-	var imageNames []string
+	newImageNames := sets.Set[string]{}
```

Comment:
nit: `newImageNames` makes me think about the "net new image", the one that have been discovered and are not yet known (as in, do not have an Image CR associated).

Can we pick a different name?